doc/sphinx/source/fusion/pattern-matching.rst

```diff
@@ -0,0 +1,81 @@
+.. fusion/pattern-matching.rst: 
+
+Applying pattern matcher to fuse ops
+-------------------------------------
+
+Before delving into the details of pattern matching, note that the sole purpose 
+of the pattern-matching step is to **find** patterns in the framework's graph of 
+outputs.  A subgraph, then, could contain any operation nGraph's IR defines 
+(addition, subtraction, etc) along with some special wildcard nodes. 
+
+An analogy that is probably familiar to many programmers is that of the ``Regex``, 
+AKA "Regular expression". Just as one can write a regex (pattern) and run it 
+through some text to find and/or replace the occurences of that pattern in 
+the given text, so too can one write optimization-passes to construct patterns 
+which are just regular nGraph graphs and run those patterns through given graphs.
+
+In the ``Regex`` analogy, 
+
+
+* Letters (for example: `A`, `B`, `m`, `n`), 
+
+* Strings (for example: `ABBA`, `Hello world`), and
+
+* Collective Symbols (for some regex programs they are `*`, `.`)
+
+
+respectfully correspond to: 
+
+.. Letter -> Node
+
+* ``ngraph::Node`` (``op::Add``, ``op::BatchNormBackprop``), 
```

Comment:
`graph nodes, such as op::Add and op::BatchNormBackprop`
It's hard to figure out that `ngraph::Node` is supposed to signify things that are graph nodes here the way it shows up.